src/main/java/org/opensearch/knn/indices/ModelGraveyard.java

```diff
@@ -151,27 +153,75 @@ public static NamedDiff readDiffFrom(StreamInput streamInput) throws IOException
+        // {model_ids:[]} - Parses and returns an empty ModelGraveyard object as there are no model ids
+        // {model_ids: ["abcd", "1234"]} - Parses and returns a ModelGraveyard object which contains the model ids "abcd" and "1234"
+        // {model_ids:[],dummy_field:[]} - Throws exception as we have FIELD_NAME(dummy_field) instead of END_OBJECT token
+
         ModelGraveyard modelGraveyard = new ModelGraveyard();
 
         // If it is a fresh parser, move to the first token
         if (xContentParser.currentToken() == null) {
             xContentParser.nextToken();
         }
 
-        // on a start object move to next token
-        if (xContentParser.currentToken() == XContentParser.Token.START_OBJECT) {
-            xContentParser.nextToken();
+        // Validate if the first token is START_OBJECT
+        if (xContentParser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new OpenSearchParseException(
+                "Unable to parse ModelGraveyard. Expecting token start of an object but got {}",
+                xContentParser.currentToken()
+            );
         }
 
+        // Adding Backward Compatibility for the domains that have already parsed the old toXContent logic which has XContent as {}
+        if (xContentParser.nextToken() == XContentParser.Token.END_OBJECT) {
+            return modelGraveyard;
+        }
+
+        // Validate it starts with FIELD_NAME token after START_OBJECT
         if (xContentParser.currentToken() != XContentParser.Token.FIELD_NAME) {
-            throw new IllegalArgumentException("expected field name but got a " + xContentParser.currentToken());
+            throw new OpenSearchParseException(
+                "Unable to parse ModelGraveyard. Expecting token field name but got {}",
+                xContentParser.currentToken()
+            );
+        }
+
+        // Validating that FIELD_NAME matches with "model_ids"
+        if (!MODEL_IDS.equals(xContentParser.currentName())) {
+            throw new OpenSearchParseException(
+                "Unable to parse ModelGraveyard. Expecting field {} but got {}",
+                MODEL_IDS,
+                xContentParser.currentName()
+            );
+        }
+
+        // Validate it starts with START_ARRAY token after FIELD_NAME
+        if (xContentParser.nextToken() != XContentParser.Token.START_ARRAY) {
+            throw new OpenSearchParseException(
+                "Unable to parse ModelGraveyard. Expecting token start of an array but got {}",
+                xContentParser.currentToken()
+            );
         }
 
-        while (xContentParser.nextToken() != XContentParser.Token.END_OBJECT) {
+        while (xContentParser.nextToken() != XContentParser.Token.END_ARRAY) {
             if (xContentParser.currentToken() == XContentParser.Token.VALUE_STRING) {
                 modelGraveyard.add(xContentParser.text());
             }
```

Comment:
Do we want to ignore or throw exception when the token is not `VALUE_STRING`?

Reply:
Added exception